VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/General/ImplementEqualsAndGetHashCode/ImplementEqualsAndGetHashCodeCodeFix.cs

```diff
@@ -0,0 +1,274 @@
+                                                 fieldAndPropertyGetHashCodeStatements) + ";")
+                    .WithLeadingTrivia(
+                        SyntaxFactory.ParseLeadingTrivia(
+@"// Add any fields you're interested in, taking into account the guidelines described in
+// https://msdn.microsoft.com/en-us/library/system.object.gethashcode%28v=vs.110%29.aspx
+"));
+
+            return SyntaxFactory.MethodDeclaration(SyntaxFactory.ParseTypeName("int"), "GetHashCode")
+                    .AddModifiers(publicModifier, overrideModifier)
+                    .AddBodyStatements(returnStatement)
+                    .WithAdditionalAnnotations(Formatter.Annotation);
+        }
+
+        private IEnumerable<string> GetFieldHashCodeStatements(SemanticModel model, SyntaxNode member)
+        {
+            if (!member.IsKind(SyntaxKind.FieldDeclaration))
+            {
+                return Enumerable.Empty<string>();
+            }
+
+            var field = (FieldDeclarationSyntax) member;
+            if (field.Modifiers.ContainsAny(SyntaxKind.StaticKeyword, SyntaxKind.ConstKeyword))
+            {
+                return Enumerable.Empty<string>();
+            }
+
+            var symbol = model.GetTypeInfo(field.Declaration.Type).Type;
+            if (symbol == null || !symbol.IsValueType && symbol.SpecialType != SpecialType.System_String)
+            {
+                return Enumerable.Empty<string>();
+            }
+
+            if (field.Modifiers.Contains(SyntaxKind.ReadOnlyKeyword))
+            {
+                return field.Declaration.Variables.Select(
+                    variable => $"{variable.Identifier}.GetHashCode()");
+            }
+
+            return Enumerable.Empty<string>();
+        }
+
+        private string GetPropertyHashCodeStatement(SemanticModel model, SyntaxNode member)
+        {
+            if (!member.IsKind(SyntaxKind.PropertyDeclaration))
+            {
+                return string.Empty;
+            }
+
+            var property = (PropertyDeclarationSyntax)member;
+            if (property.Modifiers.Contains(SyntaxKind.StaticKeyword))
+            {
+                return string.Empty;
+            }
+
+            var symbol = model.GetTypeInfo(property.Type).Type;
+            if (symbol == null || !symbol.IsValueType && symbol.SpecialType != SpecialType.System_String)
+            {
+                return string.Empty;
+            }
+
+            if (property.AccessorList.Accessors.Any(a => a.IsKind(SyntaxKind.SetAccessorDeclaration)))
+            {
+                return string.Empty;
+            }
+
+            // ensure getter does not have body
+            // the property has to have at least one of {get, set}, and it doesn't have a set (see above)
+            // this will not have an NRE in First()
+            if (property.AccessorList.Accessors.First(a => a.IsKind(SyntaxKind.GetAccessorDeclaration)).Body == null)
```

Comment:
That is considered a get-only property.  I re-wrote this, but basically, it ensures the IPropertySymbol `GetMethod` property is not null.